### ISSUE TITLE
Feature request: Krypton / https://krypt.co/ support for SSH 

### ISSUE THREAD

**Issue by ukd1** [jakob/Postico#565]
I'd love support for Krypton (cc @kryptco: @agrinman, @kcking, @ryantberg) - currently Postico errors with the following warning about not supported key type:

<img width="640" alt="screen shot 2018-06-21 at 8 40 01 am" src="https://user-images.githubusercontent.com/44345/41729693-b7e2475e-752e-11e8-8719-171f3944c964.png">

even though it's an ssh-rsa key;

```bash
macbooklols:~ russ$ cat ~/.ssh/id_krypton.pub 
ssh-rsa AAAAB3NzaC1.......................
```

<3

**Comment by agrinman** [comment]
@ukd1 It looks like Postico only supports specifying a private key from within their connection UI. In the case of Krypton, the private key only lives on your phone. My guess is that Postico uses some version of `libssh` instead of system ssh. If there was an option to use system ssh or a mechanism to specify an ssh agent (our `krssh` agent) then it should just work. 

Happy to help the Postico team to make it Krypton compatible :)

**Comment by jakob** [comment]
Indeed, Postico only supports reading private keys from files. Postico 1.4 adds support for modern keys (ecdsa, ed25519), but you still need to have the private key on disk.

Postico uses libssh2, which in theory does support ssh agents, but unfortunately sandboxed apps cannot connect to unix sockets outside of their sandbox, preventing Postico from connecting to ssh agent.

From [the docs](https://krypt.co/docs/start/installation.html), I see that krypton makes a lot of changes to the ssh config. Is ProxyCommand absolutely necessary, or will Krypton work without it? Also, why the `IdentityFile ~/.ssh/id_krypton`? Shouldn't it be possible to get public keys from the agent?

**Comment by agrinman** [comment]
@jakob The sandbox does indeed seem to prevent compatibility then. I am not sure if there are ways to add file support to break out of the sandbox...

ProxyCommand is used so we can intercept the host signature and pass it on to Krypton to do host verification. In principal you don't need it but then the user will see "unknown host". The IdentityFile is to prioritize the Krypton key -- but can work without it too.
